script.js

```diff
@@ -1,12 +1,169 @@
 //You can edit ALL of the code here
+
+let allEpisodes = getAllEpisodes();
 function setup() {
-  const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
 }
 
+// headerContainer.classList.add("header-container");
+//Level 350 - Switch to fetching live data!
+function fetchAllEpisodes(series) {
+  return fetch(`https://api.tvmaze.com/shows/${series}/episodes`)
+    .then((response) => response.json())
+    .then((data) => {
+      allEpisodes = data;
+      makePageForEpisodes(allEpisodes);
+    })
+    .catch(function (error) {
+      console.log(error);
+    });
+}
+
+function tvSeriesSetup(showList) {
+  let selectSeries = document.getElementById("show-select");
+  let optionSeries = document.createElement("option");
+  optionSeries.textContent = "Please Choose Your Show";
+  selectSeries.appendChild(optionSeries);
+  selectSeries.selectedIndex = 0;
+
+  //sorting based on alphabetical order
+  showList.sort(function (a, b) {
+    return a.name.localeCompare(b.name);
+  });
```

Comment:
Since you are sorting strings, you can use `.sort()` without a callabck function here

Reply:
Tried `showList.sort()` but didn't work so i use another alternative way as following: 
`showList.sort((a, b) => {
    if (a.name.toLowerCase() > b.name.toLowerCase()) {
      return 1;
    }
    if (a.name.toLowerCase() < b.name.toLowerCase()) {
      return -1;
    }
    return 0;
  });`

Reply:
which is working